v2/http-middleware/security.go

```diff
@@ -211,11 +211,8 @@ func AuthorizeMiddleware(authorizer Authorizer) mux.MiddlewareFunc {
 					WithField("url", req.URL.String()).
 					Error("Couldn't extract User ID from context")
 
-				responseBody := GetInternalServerErrorResponseBody(http_model.ErrResponseInternalServerError, secConfig)
-
-				http_server.WriteJSONResponse(
-					ctx, w, req, http.StatusInternalServerError, responseBody,
-				)
+				responseBody := GetUnauthenticedErrorResponseBody(http_model.ErrResponseUnauthorized, secConfig)
```

Comment:
One might argue that not using the Authentication middleware before Authorize middleware is an internal error and nothing the user can do.

To me, that sounds like a 500 Internal Server Error?